url2698a2c731_cassandra_project_cassandra_stand_alone_tgz-pJ8-eu_cassandra_sim_MySimpleDynamicScenarioSimulationJ8/src/eu/cassandra/sim/entities/appliances/ConsumptionModel.java

```diff
@@ -0,0 +1,358 @@
+
+	/**
+	 * Gets the number of times that the i-th pattern runs.
+	 *
+	 * @param i the index i
+	 * @return the the number of times that the i-th pattern runs
+	 */
+	public int getN(int i) { return n[i]; }
+
+	/**
+	 * Gets the i-th pattern's duration.
+	 *
+	 * @param i the index i
+	 * @return the i-th pattern's duration
+	 */
+	public int getPatternDuration(int i) { return patternDuration[i]; }
+
+	/**
+	 * Gets the i-th pattern.
+	 *
+	 * @param i the index i
+	 * @return the i-th pattern
+	 */
+	public ArrayList<Tripplet> getPattern(int i) { return patterns[i]; }
+
+	/**
+	 * Gets the array storing the consumption patterns.
+	 *
+	 * @return the array storing the consumption patterns
+	 */
+	public ArrayList<Tripplet>[] getPatterns() { return patterns; }
+
+	/**
+	 * Check if consumption is static.
+	 *
+	 * @return true, if consumption is static.
+	 */
+	public boolean checkStatic()
+	{
+		boolean result = true;
+
+		Double[] values = getValues();
+
+		// System.out.println("Appliance: " + name + " Model: "
+		// + activeConsumptionModelString);
+		//
+		// System.out.println("Appliance: " + name + " Values: "
+		// + Arrays.toString(values));
+
+		for (int i = 0; i < values.length - 1; i++) {
+			// System.out.println("Previous: " + values[i].doubleValue() + " Next: "
+			// + values[i + 1].doubleValue());
+			if (values[i].doubleValue() != values[i + 1].doubleValue()) {
+				// System.out.println("IN");
+				result = false;
+				break;
+			}
+		}
+
+		return result;
+	}
+
+	/**
+	 * Gets the consumption.
+	 *
+	 * @return the consumption
+	 */
+	public Double[] getConsumption()
+	{
+
+		ArrayList<Double> temp = new ArrayList<Double>();
+		int times = getOuterN();
+		if (times == 0)
+			times = 2;
+		// Number of repeats
+		for (int i = 0; i < times; i++) {
+			// System.out.println("Time: " + i);
+			// Number of patterns in each repeat
+			for (int j = 0; j < getPatternN(); j++) {
+				// System.out.println("Pattern: " + j);
+				int internalTimes = getN(j);
+				if (internalTimes == 0)
+					internalTimes = 2;
+				// System.out.println("Internal Times: " + k);
+				for (int k = 0; k < internalTimes; k++) {
+					ArrayList<Tripplet> tripplets = getPattern(j);
+					for (int l = 0; l < tripplets.size(); l++) {
+						// System.out.println("TripletPower: " + l);
+						for (int m = 0; m < tripplets.get(l).d; m++) {
+							temp.add(tripplets.get(l).v);
+						}
+					}
+				}
+			}
+		}
+		Double[] result = new Double[temp.size()];
+		temp.toArray(result);
+		return result;
+
+	}
+
+	/**
+	 * Gets the consumption model as a BasicDBObject.
+	 *
+	 * @return the consumption model as a BasicDBObject
+	 */
+	public BasicDBObject toDBObject() {
+		BasicDBObject obj = new BasicDBObject();
+		obj.put("name", name);
+		obj.put("description", description);
+		obj.put("app_id", parentId);
+		obj.put("model", model);
+		return obj;
+	}
+
+
+	private void init (DBObject modelObj, String type) throws Exception {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 19. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.